src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.TagSource_ProduceTags.cs

```diff
@@ -617,23 +607,27 @@ private Task ProduceTagsAsync(TaggerContext<TTag> context, CancellationToken can
             if (_disposalTokenSource.Token.IsCancellationRequested)
                 return null;
 
-            // If this is the first time we're being asked for tags, and we're a tagger that requires the initial
-            // tags be available synchronously on this call, and the computation of tags hasn't completed yet, then
-            // force the tags to be computed now on this thread.  The singular use case for this is Outlining which
-            // needs those tags synchronously computed for things like Metadata-as-Source collapsing.
+            // If this is the first time we're being asked for tags, and we're a tagger that requires the initial tags
+            // be available synchronously on this call, and the computation of tags hasn't completed yet, then force the
+            // tags to be computed now on this thread.  The singular use case for this is Outlining which needs those
+            // tags synchronously computed for things like Metadata-as-Source collapsing.
+            var tagTrees = _cachedTagTrees_mayChangeFromAnyThread;
```

Comment:
to keep the logic sane here, we grab whatever is cached and use that.  if we see we're forced to compute up to date tags, we call into the helper, with our JTF.Run call below.  That helper returns the final tag trees it produced, which we then use to answer the call.